internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -395,16 +401,12 @@ func (t *Translator) translateBackendTrafficPolicyForRoute(policy *egv1a1.Backen
 						BackendConnection: bc,
 						HTTP2:             h2,
 						DNS:               ds,
+						Timeout:           to,
 					}
 
 					// Update the Host field in HealthCheck, now that we have access to the Route Hostname.
 					r.Traffic.HealthCheck.SetHTTPHostIfAbsent(r.Hostname)
 
-					// Some timeout setting originate from the route.
```

Comment:
shouldnt this have worked as well ?

Reply:
The problem was an ordering problem. The timeout value was lost in the original ordering because `r.Traffic` (which contained the value collected from the `HTTPRoute` was overridden in line 392.